src/commands/createNewProject/ProjectCreateStep/GradleProjectCreateSteps.ts

```diff
@@ -0,0 +1,174 @@
+        return new GradleProjectCreateStep();
+    }
+
+    public async executeCore(context: IJavaProjectWizardContext, progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
+        await super.executeCore(context, progress);
+
+        const settingsGradlePath: string = path.join(context.projectPath, settingsGradleFileName);
+        if (await confirmOverwriteFile(context, settingsGradlePath)) {
+            await fse.writeFile(settingsGradlePath, this.getSettingsGradleContent(context));
+        }
+
+        const buildGradlePath: string = path.join(context.projectPath, buildGradleFileName);
+        const buildGradleContent: string = await this.getBuildGradleContent(context);
+        if (await confirmOverwriteFile(context, buildGradlePath)) {
+            await fse.writeFile(buildGradlePath, buildGradleContent);
+        }
+    }
+
+    public shouldExecute(context: IJavaProjectWizardContext): boolean {
+        return context.buildTool === JavaBuildTool.gradle;
+    }
+
+    async getLatestGradlePluginVersion(context: IJavaProjectWizardContext): Promise<string> {
+        try {
+            const templateVersion: string | undefined = await javaUtils.getLatestArtifactVersionFromMetaData(context, metaDataUrl);
+            return templateVersion ? templateVersion : backupGradlePluginVersion;
+        } catch (error) {
+            return backupGradlePluginVersion;
+        }
+    }
+
+    getSettingsGradleContent(context: IJavaProjectWizardContext): string {
+        return `rootProject.name = "${context.javaArtifactId}"`;
+    }
+
+    getCompatibilityVersion(javaVersion: string | undefined): string {
+        if (javaVersion === java8) {
+            return "1.8";
+        } else if (javaVersion === java11) {
+            return "11";
+        } else {
+            throw new Error(localize('invalidJavaVersion', 'Invalid Java version "{0}".', javaVersion));
+        }
+    }
+
+    async getBuildGradleContent(context: IJavaProjectWizardContext): Promise<string> {
+        return `plugins {
+  id "com.microsoft.azure.azurefunctions" version "${await this.getLatestGradlePluginVersion(context)}"
+}
+apply plugin: 'java'
+apply plugin: "com.microsoft.azure.azurefunctions"
+
+group '${context.javaGroupId}'
+version '${context.javaProjectVersion}'
+
+dependencies {
+    implementation 'com.microsoft.azure.functions:azure-functions-java-library:1.4.2'
+    testImplementation 'org.junit.jupiter:junit-jupiter:5.6.2'
+    testImplementation 'org.mockito:mockito-core:3.3.3'
+}
+
+sourceCompatibility = '${this.getCompatibilityVersion(context.javaVersion)}'
+targetCompatibility = '${this.getCompatibilityVersion(context.javaVersion)}'
+
+compileJava.options.encoding = 'UTF-8'
+
+repositories {
+    mavenCentral()
+}
+
+azurefunctions {
+    resourceGroup = 'java-functions-group'
+    appName = '${context.javaAppName}'
+    pricingTier = 'Consumption'
+    region = 'westus'
```

Comment:
This is the default value for function gradle plugin, refers [here](https://github.com/Azure-Samples/azure-functions-samples-java/blob/master/build.gradle). Users could modify the value after the project was created if necessary, and this will not affect the behavior of vscode extension.